docs/user-defined-type-functions.md

```diff
@@ -0,0 +1,235 @@
+| `getboolean()` | `typelib` | returns an immutable runtime representation of the built-in type `boolean` |
+| `getnumber()` | `typelib` | returns an immutable runtime representation of the built-in type `number` |
+| `getstring()` | `typelib` | returns an immutable runtime representation of the built-in type `string` |
+| `getstringsingleton(arg: string)` | `typelib` | returns an immutable runtime representation of a string singleton type of the argument |
+| `getbooleansingleton(arg: boolean)` | `typelib` | returns an immutable runtime representation of a boolean singleton type of the argument |
+| `getunion(arg: {typelib})` | `typelib` | returns an immutable runtime representation of union type of its argument |
+| `getintersection(arg: {typelib})` | `typelib` | returns an immutable runtime representation of intersection type of its argument |
+| `newtable()` | `typelib` | returns a mutable runtime representation of a `table` type |
+| `newmetatable()` | `typelib` | returns a mutable runtime representation of `Metatable` |
+| `newfunction()` | `typelib` | returns a mutable runtime representation of a `function` type |
+| `isnil()` | `boolean` | returns true if self is a runtime representation of the built-in type `nil` |
+| `isunknown()` | `boolean` | returns true if self is a runtime representation of the built-in type `unknown` |
+| `isnever()` | `boolean` | returns true if self is a runtime representation of the built-in type `never` |
+| `isany()` | `boolean` | returns true if self is a runtime representation of the built-in type `any` |
+| `isnegation()` | `boolean` | returns true if self is a runtime representation of a `Negation` |
+| `isboolean()` | `boolean` | returns true if self is a runtime representation of the built-in type`boolean` |
+| `isnumber()` | `boolean` | returns true if self is a runtime representation of the built-in type `number` |
+| `isstring()` | `boolean` | returns true if self is a runtime representation of the built-in type `string` |
+| `isstringsingleton()` | `boolean` | returns true if self is a runtime representation of a string singleton |
+| `isbooleansingleton()` | `boolean` | returns true if self is a runtime representation of a boolean singleton |
+| `isunion()` | `boolean` | returns true if self is a runtime representation of the union type |
+| `isintersection()` | `boolean` | returns true if self is a runtime representation of the intersection type |
+| `istable()` | `boolean` | returns true if self is a runtime representation of a `table` type |
+| `ismetatable()` | `boolean` | returns true if self is a runtime representation of `Metatable` |
+| `isfunction()` | `boolean` | returns true if self is a runtime representation of a `function` type |
+| `isclass()` | `boolean` | returns true if self is a runtime representation of a `class` type |
+| `issubtypeof(arg: typelib)` | `boolean` | returns true if self is a subtype or equal to arg in the type hierarchy |
```

Comment:
I'm not necessarily averse to having this in the API, but it's worth noting that this essentially requires that this library depend on `Normalization` or provides its own because Luau's notion of subtyping is fundamentally semantic. That is probably going to be a huge blocker, and this is, by far, the most complicated thing to implement in this RFC.